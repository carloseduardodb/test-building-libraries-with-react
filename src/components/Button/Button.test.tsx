// import renderer from 'react-test-renderer';
// import { Button } from './index';

// describe('Button', () => {
//   it('renders correctly', () => {
//     const tree = renderer.create(<Button />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('test_button_renders_with_custom_className_prop', () => {
//     const tree: any = renderer
//       .create(<Button className="custom-class" />)
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//     expect(tree.props.className).toEqual('custom-class');
//   });

//   it('test_button_renders_with_default_text', () => {
//     const tree = renderer.create(<Button>Click-Me</Button>).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('test_button_renders_with_disabled_prop', () => {
//     const tree: any = renderer.create(<Button disabled />).toJSON();
//     expect(tree).toMatchSnapshot();
//     expect(tree.props.disabled).toEqual(true);
//   });

//   it('test_button_renders_with_type_prop', () => {
//     const tree: any = renderer.create(<Button type="submit" />).toJSON();
//     expect(tree).toMatchSnapshot();
//     expect(tree.props.type).toEqual('submit');
//   });
// });
