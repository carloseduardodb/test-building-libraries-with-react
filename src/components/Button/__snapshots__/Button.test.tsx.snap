// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `<button />`;

exports[`Button test_button_renders_with_custom_className_prop 1`] = `
<button
  className="custom-class"
/>
`;

exports[`Button test_button_renders_with_default_text 1`] = `
<button>
  Click-Me
</button>
`;

exports[`Button test_button_renders_with_disabled_prop 1`] = `
<button
  disabled={true}
/>
`;

exports[`Button test_button_renders_with_type_prop 1`] = `
<button
  type="submit"
/>
`;
